SignalShareExtension/ShareViewController.swift

```diff
@@ -51,7 +51,11 @@ public class ShareViewController: UINavigationController, SAELoadViewDelegate {
 
         isReadyForAppExtensions = OWSPreferences.isReadyForAppExtensions()
         if !isReadyForAppExtensions {
-            // TODO: Show the "You need to launch main app" view.
+            if (OWSPreferences.isRegistered()) {
```

Comment:
TIOLI:

I might be wrong, but I'm guessing this state is being duplicated from TSStorageManagaer so that we can show "Not Registered" immediately, rather than briefly showing "Loading..." before showing "Not Registered". Is that right?

I think in practice, if the user is not registered, the db should set up *really* quickly, so we wouldn't be waiting for long before showing "Not Registered". But maybe this produces an undesirable flicker while the "Loading..." is shown and removed. 

What I'm getting at, is that I prefer my hacks at the UI layer rather than the persistence layer. Maybe there's something we can do to address the flicker issue without duplicating state.

**Pedantry aside**, I only have one specific concern: Will a user who restores signal from a backup and launches the share extension before re-registering be in a confused state due to their out-of-sync NSUserDefaults?

The current consequences of which wouldn't be severe, but it would be easy to use this preference elsewhere in a more dangerous place.

Reply:
> I might be wrong, but I'm guessing this state is being duplicated from TSStorageManagaer so that we can show "Not Registered" immediately, rather than briefly showing "Loading..." before showing "Not Registered". Is that right?

Nope.  Sorry, I should have added comments here.  Done.